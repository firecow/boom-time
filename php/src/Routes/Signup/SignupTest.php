<?
declare(strict_types=1);

namespace App\Routes\Signup;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class SignupTest extends RegressionTest
{

    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPPostValue("username", "immabad");
        $superGlobals->setHTTPPostValue("password", "immaverybadpassword");
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/signup/");
        return $routeResponse->prettyPrint();
    }
}