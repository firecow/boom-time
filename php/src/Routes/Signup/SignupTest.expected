HTTP/1.1 200 Ok
Content-Length: 258
Content-Type: application/json

bearerToken=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTEyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6IjZjOGUxNDVmYzdiNDBiNzZmYTc4MzIyMGE0OGZlZDlhIiwiaW1nIjpudWxsLCJ1bmEiOiJpbW1hYmFkIn0.3uucgfBLx9h9CT0gS5xuM8OqKzZGanFgZnCp2ib5tEQ
expires=Friday, 23-Feb-2018 13:02:02 UTC

{
    "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTEyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6IjZjOGUxNDVmYzdiNDBiNzZmYTc4MzIyMGE0OGZlZDlhIiwiaW1nIjpudWxsLCJ1bmEiOiJpbW1hYmFkIn0.3uucgfBLx9h9CT0gS5xuM8OqKzZGanFgZnCp2ib5tEQ"
}

+ INSERT INTO `user_passwords` VALUES ('6c8e145fc7b40b76fa783220a48fed9a','8b40a04dabd100b04b26290940cb2782');
+ INSERT INTO `users` VALUES ('6c8e145fc7b40b76fa783220a48fed9a','immabad',NULL);
